File: backend/src/ai/explainEngine.js
/**
 * Generates an explanation array based on the provided data.
 *
 * @param {Object} data - The input data object.
 * @param {string} data.item - The detected item.
 * @param {number} data.coins - The earned EcoCoins.
 * @param {number} data.trustScore - The user's Trust Score.
 * @param {string} data.priority - The pickup priority.
 *
 * @returns {string[]} An array of explanation strings.
 */
function generateExplanation(data) {
    // Extract the input data from the data object
    const { item, coins, trustScore, priority } = data;
  
    // Create an empty array to store the explanation strings
    const explanation = [];
  
    // Add the detected item to the explanation
    explanation.push(`${item} detected.`);
  
    // Add the earned EcoCoins to the explanation
    explanation.push(`You earned ${coins} EcoCoins.`);
  
    // Add the user's Trust Score to the explanation
    explanation.push(`Your Trust Score is ${trustScore}.`);
  
    // Add the pickup priority to the explanation
    explanation.push(`Pickup Priority: ${priority}.`);
  
    // Return the explanation array
    return explanation;
  }
  
  // Export the generateExplanation function using CommonJS
  module.exports = generateExplanation;

File: backend/src/ai/explain_engine.py
def generate_explanation(data):
    """
    Generates a list of user-friendly messages based on the provided data.

    Args:
        data (dict): A dictionary containing the data for explanation.
            - item (str): The detected item.
            - coins (int): The number of EcoCoins earned.
            - trustScore (int): The user's trust score.
            - priority (str): The pickup priority.

    Returns:
        list: A list of user-friendly messages.
    """
    explanation = []

    item_message = f"{data['item']} detected."
    explanation.append(item_message)

    coins_message = f"You earned {data['coins']} EcoCoins."
    explanation.append(coins_message)

    trust_score_message = f"Your Trust Score is {data['trustScore']}."
    explanation.append(trust_score_message)

    priority_message = f"Pickup Priority: {data['priority']}."
    explanation.append(priority_message)

    return explanation


if _name_ == "_main_":
    test_data = {
        "item": "Plastic Bottle",
        "coins": 30,
        "trustScore": 82,
        "priority": "Medium"
    }

    result = generate_explanation(test_data)
    for message in result:
        print(message)